code/datums/attack_style/unarmed_damage.dm

```diff
@@ -0,0 +1,225 @@
+		return 0
+
+	return miss_chance_modifier
+
+/datum/attack_style/unarmed/generic_damage/finalize_attack(mob/living/attacker, mob/living/smacked, obj/item/bodypart/weapon, right_clicking)
+	var/damage = max(0, select_damage(attacker, smacked, weapon))
+	var/attack_verb = select_attack_verb(attacker, smacked, weapon, damage)
+	var/damage_type = attack_type || attacker.get_attack_type()
+
+	if(smacked.check_block(attacker, damage, "[attacker]'s [default_attack_verb]", UNARMED_ATTACK, attack_penetration_modifier, damage_type))
+		return ATTACK_SWING_BLOCKED
+
+	// Todo : move this out and into its own style?
+	if(!HAS_TRAIT(smacked, TRAIT_MARTIAL_ARTS_IMMUNE))
+		var/datum/martial_art/art = attacker.mind?.martial_art
+		switch(art?.harm_act(attacker, smacked))
+			if(MARTIAL_ATTACK_SUCCESS)
+				return ATTACK_SWING_HIT
+			if(MARTIAL_ATTACK_FAIL)
+				return ATTACK_SWING_MISSED
+
+	var/obj/item/bodypart/affecting = smacked.get_bodypart(smacked.get_random_valid_zone(attacker.zone_selected))
+	var/miss_chance = calculate_miss_chance(attacker, smacked, weapon, damage)
+
+	if(damage <= 0 || (iscarbon(smacked) && !istype(affecting)) || prob(miss_chance))
+		smacked.visible_message(
+			span_danger("[attacker]'s [attack_verb] misses [smacked]!"),
+			span_danger("You avoid [attacker]'s [attack_verb]!"),
+			span_hear("You hear [deaf_miss_phrase]!"),
+			vision_distance = COMBAT_MESSAGE_RANGE,
+			ignored_mobs = attacker,
+		)
+		to_chat(attacker, span_warning("Your [attack_verb] misses [smacked]!"))
+		log_combat(attacker, smacked, "missed unarmed attack ([attack_verb])")
+		return ATTACK_SWING_MISSED
+
+	// All unarmed attacks go under melee armor, even ones that use non-standard damage types like burn
+	var/armor_block = min(ARMOR_MAX_BLOCK, smacked.run_armor_check(affecting, MELEE, armour_penetration = attack_penetration_modifier))
+
+	// melbert todo : probably sounds silly
+	smacked.visible_message(
+		span_danger("[attacker] [attack_verb]ed [smacked]!"),
+		span_userdanger("You're [attack_verb]ed by [attacker]!"),
+		span_hear("You hear a sickening sound of flesh hitting flesh!"),
+		vision_distance = COMBAT_MESSAGE_RANGE,
+		ignored_mobs = attacker,
+	)
+	to_chat(attacker, span_danger("You [default_attack_verb] [smacked]!"))
+
+	UPDATE_LAST_ATTACKER(smacked, attacker)
+
+	// A damage packet is created to pass onto [actually_apply_damage].
+	// This allows for subtypes to overide certain facets of the outgoing damage if they prefer.
+	var/datum/apply_damage_packet/packet = new(
+		/* damage = */damage,
+		/* damagetype = */damage_type,
+		/* def_zone = */affecting,
+		/* blocked = */armor_block,
+		/* forced = */FALSE,
+		/* spread_damage = */FALSE,
+		/* wound_bonus = */wound_bonus_modifier,
+		/* bare_wound_bonus = */bare_wound_bonus_modifier,
+		/* sharpness = */NONE,
+		/* attack_direction = */get_dir(attacker, smacked),
+		/* attack attacking_item = */null,
+	)
```

Comment:
little uncomfortable with this but I am likely just being paranoid

Reply:
I'm not too happy with this either, but I wanted a compile-time way of ensuring correct args are passed to `apply_damage` through child implementations of `actually_apply_damage`, and arglist didn't seem to really cut it.

If you have an alternative I'm all ears